File: src/test/java/htsjdk/samtools/BamFileIoUtilsTest.java
package htsjdk.samtools;

import org.testng.Assert;
import org.testng.annotations.Test;

import java.io.File;

public class BamFileIoUtilsTest {
    @Test
    public void testIsBamFile() {
        Assert.assertTrue(BamFileIoUtils.isBamFile(new File("test.bam")));
        Assert.assertTrue(BamFileIoUtils.isBamFile(new File("test.sam.bam")));
        Assert.assertTrue(BamFileIoUtils.isBamFile(new File(".bam")));
        Assert.assertTrue(BamFileIoUtils.isBamFile(new File("./test.bam")));
        Assert.assertTrue(BamFileIoUtils.isBamFile(new File("./test..bam")));
        Assert.assertTrue(BamFileIoUtils.isBamFile(new File("c:\\path\\to\test.bam")));

        Assert.assertFalse(BamFileIoUtils.isBamFile(new File("test.Bam")));
        Assert.assertFalse(BamFileIoUtils.isBamFile(new File("test.BAM")));
        Assert.assertFalse(BamFileIoUtils.isBamFile(new File("test.sam")));
        Assert.assertFalse(BamFileIoUtils.isBamFile(new File("test.bam.sam")));
        Assert.assertFalse(BamFileIoUtils.isBamFile(new File("testbam")));
    }
}

File: src/test/scala/htsjdk/samtools/BamFileIoUtilsTest.scala
package htsjdk.samtools

import java.io.File
import htsjdk.UnitSpec

class BamFileIoUtilsTest extends UnitSpec {

  "BamFileIoUtils.isBamFile" should "identify bam files" in {
    Seq(
      ("test.bam", true),
      ("test.sam.bam", true),
      (".bam", true),
      ("./test.bam", true),
      ("./test..bam", true),
      ("c:\\path\\to\test.bam", true),
      ("test.Bam", false),
      ("test.BAM", false),
      ("test.sam", false),
      ("test.bam.sam", false),
      ("testbam", false)).foreach {
      case (file, isBam) =>
        BamFileIoUtils.isBamFile(new File(file)) shouldEqual isBam
    }
  }
}

